apps/src/StudioApp.js

```diff
@@ -479,12 +479,25 @@ StudioApp.prototype.init = function (config) {
   this.alertIfAbusiveProject('#codeWorkspace');
 
   if (this.share && config.shareWarningInfo) {
-    shareWarnings.checkSharedAppWarnings({
-      channelId: config.channel,
-      isSignedIn: config.isSignedIn,
-      hasDataAPIs: config.shareWarningInfo.hasDataAPIs,
-      onWarningsComplete: config.shareWarningInfo.onWarningsComplete
-    });
+    var showWarnings = function () {
+      shareWarnings.checkSharedAppWarnings({
+        channelId: config.channel,
+        isSignedIn: config.isSignedIn,
+        hasDataAPIs: config.shareWarningInfo.hasDataAPIs,
+        onWarningsComplete: config.shareWarningInfo.onWarningsComplete
+      });
+    };
+    if (config.level.iframeEmbed) {
+      // temporarily replace startIFrameEmbeddedApp with a call to showWarnings
+      // this is kind of wonky IMO, would appreciate suggestions for how I ought
+      // to do this.
+      this.startIFrameEmbeddedApp = (originalFunc => function () {
+        showWarnings();
+        this.startIFrameEmbeddedApp = originalFunc;
+      })(this.startIFrameEmbeddedApp);
+    } else {
```

Comment:
`startIFrameEmbeddedApp` receives `config` as a parameter.  Why not make `showWarnings` take `config` as a parameter too, and define it elsewhere?  Then wrap it in something like [`_.once`](https://lodash.com/docs#once) and call it at the top of the normal `startIFrameEmbeddedApp`.

Not sure it's much _better_ but it's flatter.